### PR TITLE
TINKERPOP-3023 Addeed Set syntax in grammar

### PR DESCRIPTION
**Pull request by spmallette**
https://issues.apache.org/jira/browse/TINKERPOP-3023

Updated translators and serializers as needed. For the latter, particular attention was needed for Javascript where Set was being serialized as array essentially.

VOTE +1

### CODE REVIEW

File: gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/grammar/DefaultGremlinBaseVisitor.java [hunk 1]
```diff
@@ -1728,6 +1728,9 @@ protected void notImplemented(final ParseTree ctx) {
 	/**
 	 * {@inheritDoc}
 	 */
-	@Override
-	public T visitTraversalBarrier(final GremlinParser.TraversalBarrierContext ctx) { notImplemented(ctx); return null; }
+	@Override public T visitTraversalBarrier(final GremlinParser.TraversalBarrierContext ctx) { notImplemented(ctx); return null; }
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override public T visitGenericLiteralSet(final GremlinParser.GenericLiteralSetContext ctx) { return null; }
```
Comment (vkagamlyk) [inline gremlin-core/src/main/java/org/apache/tinkerpop/gremlin/language/grammar/DefaultGremlinBaseVisitor.java:1735]:
```suggestion
	@Override public T visitGenericLiteralSet(final GremlinParser.GenericLiteralSetContext ctx) { notImplemented(ctx); return null; }
```

**Review by vkagamlyk — COMMENTED** [review]
thanks for adding Set to Grammar .
I left a couple of small nits, overall VOTE+1

File: gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/language/grammar/GeneralLiteralVisitorTest.java [hunk 2]
```diff
@@ -629,6 +633,52 @@ public void shouldParseEmptyGenericLiteralCollection() {
         }
     }
 
+    public static class ValidSetLiteralTest {
+
+        @Test
+        public void shouldParseGenericLiteralSet() {
```
[the remark this answers is not in the document]
Comment (spmallette) [reply gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/language/grammar/GeneralLiteralVisitorTest.java:639]: were you thinking something other than what is there now? it has a set in a set currently...

[the remark this answers is not in the document]
Comment (vkagamlyk) [reply gremlin-core/src/test/java/org/apache/tinkerpop/gremlin/language/grammar/GeneralLiteralVisitorTest.java:639]: I mean combination of different nested collections, like 
`"{'world', 165, 14.5, [12L, 0xA, 14.5, \"hello\"]}"`
`"['world', 165, 14.5, {12L, 0xA, 14.5, \"hello\"}]"`
`"['value': {12L, 0xA, 14.5, \"hello\"}]"`
`"[ {'hello'}: {'world'}]"`
`"{ ['value': {12L, 0xA, 14.5, \"hello\"}] }"`

Similar to what we have for map in `ValidMapLiteralTest`

File: gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/sideEffect/Inject.feature [hunk 3]
```diff
@@ -217,4 +217,28 @@ Feature: Step - inject()
...
+    Then the result should be unordered
+      | result |
+      | l[d[1].i,d[3].i,d[100].i,d[300].i] |
+
+  @GraphComputerVerificationInjectionNotSupported
+  Scenario: g_injectX1_3_100_300X_set
+    Given the modern graph
+    And the traversal of
+      """
+      g.inject({1, 3, 100, 300})
```
Comment (Cole-Greer) [inline gremlin-test/src/main/resources/org/apache/tinkerpop/gremlin/test/features/sideEffect/Inject.feature:239]: Do we have defined semantics for duplicate elements passed in the set initializer? My expectation is that they just get dropped. If so could we add a test for
```
g.inject({1, 1}) => {1}
```

**Review by Cole-Greer — APPROVED** [review]
Thanks Stephen. LGTM. Just one extra test case I want to consider.

VOTE +1

**Pull request merged**